source/common/stream_info/utility.h

```diff
@@ -119,5 +121,82 @@ class Utility {
   formatDownstreamAddressJustPort(const Network::Address::Instance& address);
 };
 
+// Static utils for creating, consuming, and producing strings from the
+// Proxy-Status HTTP response header.
+class ProxyStatusUtils {
+public:
+  // Returns a Proxy-Status request header string, of the form:
+  //
+  //     <server_name>; error=<error_type>; details=<details>
+  //
+  // where:
+  //   - node_id     is either the method argument, or the name of the proxy
+  //                 in |node_id|,
+  //   - error       is the error in |error|,
+  //   - details     is |stream_info.responseCodeDetails()|, but the field is
+  //                 present only if configured in |proxy_status_config|.
+  static const std::string
```

Comment:
C++ nit: avoid top-level const in declarations since they does nothing (throughout the file)

https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html

Reply:
Done.